### PR TITLE
Introduce new Pricesheet operation. Update existing Usage, Reservation Summaries and Reservation Details commands.

### PR DESCRIPTION
**Pull request by shbha**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5542
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py [hunk 1]
```diff
@@ -32,41 +51,84 @@ def cli_consumption_list_usage(cmd, client, billing_period_name=None, top=None,
...
     filter_expression = None
     filter_from = "properties/UsageDate ge {}".format(start_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
     filter_to = "properties/UsageDate le {}".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
     filter_expression = "{} and {}".format(filter_from, filter_to)
-    return list(client.list(scope, filter=filter_expression))
+    return list(client.list_by_reservation_order_and_reservation(reservationorderid, reservationid, filter=filter_expression))
+
+
+def cli_consumption_list_pricesheet_get(client, include_additional_properties=False, include_meter_details=False):
+    """get the pricesheet of the subscription"""
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: Don't include docstrings. Instead, add the command help summary to `help.py`.

**Comment by ms-premp** [comment]
@tjprescott looks like the travic build is failing for modules, which we do not own... Is this something we need to worry about as if it's because of this PR or this is something we don't need to worry about ?

**Comment by shbha** [comment]
@tjprescott - I'm seeing a common build-verification error "JSON is invalid: Extra data: line 1 column 2 - line 1 column 229720 (char 1 - 229719)" across the failed tests. I didnt see this error running the tests locally - would you please assist with what can be done to fix this?

**Comment by tjprescott** [comment]
@shbha thank you. Yes this is caused by bug #4127.  If you do a global search for `enable_large_payload` you'll find workarounds that will allow you to continue until we fix the test base. cc/ @yugangw-msft @troydai.

**Comment by shbha** [comment]
@tjprescott : Travis -The latest build failure is also due to network module.

**Comment by derekbekoe** [comment]
@shbha Try rebasing your changes against the latest on dev branch. This should resolve the version errors.

**Comment by shbha** [comment]
@derekbekoe - Thanks Derek, that was it.

**Comment by shbha** [comment]
Can we merge this PR today if possible thank you.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py [hunk 2]
```diff
@@ -14,15 +14,42 @@ def load_arguments(self, _):
...
+        c.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
         c.argument('billing_period_name', options_list=['--billing-period-name', '-p'], help='name of a specific billing period to get the usage details that associate with')
+
     with self.argument_context('consumption reservations summaries list') as rs:
+        rs.argument('reservationorderid', options_list=['--reservation-order-id', '-r'], help='Reservation order id')
+        rs.argument('start_date', options_list=['--start-date', '-s'], type=get_datetime_type(), help='start date (in UTC Y-m-d) of the reservation summaries. Only needed for daily grain and both start date and end date need to be supplied or neither')
+        rs.argument('end_date', options_list=['--end-date', '-e'], type=get_datetime_type(), help='end date (in UTC Y-m-d) of the reservation summaries. Only needed for daily grain and both start date and end date need to be supplied or neither')
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py]: You might consider putting these three parameters under the 'consumption reservations' context, as they're repeated across many commands.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py [hunk 3]
```diff
@@ -32,41 +47,68 @@ def cli_consumption_list_usage(cmd, client, billing_period_name=None, top=None,
...
+    return list(client.list_by_billing_period(billing_period_name, expand=expand, filter=filter_expression))
 
 
-def cli_consumption_list_reservations_summaries(client, grain, reservationorderid, reservationid=None, start_date=None, end_date=None):
-    """List all the reservation summaries """
-    if reservationid:
-        scope = "providers/Microsoft.Capacity/reservationorders/{}/reservations/{}".format(reservationorderid, reservationid)
-    else:
-        scope = "providers/Microsoft.Capacity/reservationorders/{}".format(reservationorderid)
+def cli_consumption_list_reservations_summaries(client, grain, reservationorderid, start_date=None, end_date=None):
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: You should change 'reservationorderid' to 'reservation_order_id'. It's more readable, and will also adding the parameter option '--reservation-order-id' by default.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py [hunk 4]
```diff
@@ -46,3 +46,22 @@ def reservation_details_output(result):
...
+    return result
+
+
+def pricesheet_get_output(result):
+    result.pricesheets = [pricesheet_get_properties(item) for item in result.pricesheets]
+    return result
+
+
+def transform_pricesheet_get_output(result):
+    return pricesheet_get_output(result)
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py]: why not combine transform_pricesheet_get_output() and pricesheet_get_output()?

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py (hunk 2 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py]: Or, if you are going to repeat them, they should be bundled as a CLIArgumentType so that if you later change them, you can change them in just one place.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py [hunk 5]
```diff
@@ -46,3 +46,22 @@ def reservation_details_output(result):
...
     return [reservation_details_output(item) for item in result]
+
+
+def pricesheet_get_properties(result):
+    result.unit_of_measure = str(result.unit_of_measure)
+    result.included_quantity = str(result.included_quantity)
+    result.unit_price = str(result.unit_price)
+    if result.meter_details:
+        result.meter_details.total_included_quantity = str(result.meter_details.total_included_quantity)
+        result.meter_details.pretax_standard_rate = str(result.meter_details.pretax_standard_rate)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py]: Why is this necessary?  It looks like it is simply calling `str(...)` on every property.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py (hunk 4 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py]: Either way is fine here.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py [hunk 6]
```diff
@@ -30,6 +30,15 @@ def validate_both_start_end_dates(namespace):
         raise CLIError("usage error: Both --start-date and --end-date need to be supplied or neither.")
 
 
+def validate_usage_bp_inputs(namespace):
+    """Validates the existence of both start and end dates in the parameter or neither"""
+    if (namespace.start_date and not namespace.end_date) or (namespace.end_date and not namespace.start_date):
+        raise CLIError("usage error: Both --start-date and --end-date need to be supplied or neither.")
+
+    if not namespace.billing_period_name:
+        raise CLIError("usage error: Billing Period name --billing-period-name needs to be supplied")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py]: This should not be needed. Argparse will detect any missing required parameters.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py [hunk 7]
```diff
@@ -30,6 +30,15 @@ def validate_both_start_end_dates(namespace):
         raise CLIError("usage error: Both --start-date and --end-date need to be supplied or neither.")
 
 
+def validate_usage_bp_inputs(namespace):
+    """Validates the existence of both start and end dates in the parameter or neither"""
+    if (namespace.start_date and not namespace.end_date) or (namespace.end_date and not namespace.start_date):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py]: Recommend the more concise: 
`if bool(namespace.start_date) != bool(namespace.end_date):`

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py [hunk 8]
```diff
@@ -41,6 +50,18 @@ def validate_reservations_summaries(namespace):
...
+def validate_reservations_details_reservationid(namespace):
+    """Validates reservation order id """
+    if not namespace.reservationorderid:
+        raise CLIError("usage error: Reservation Order Id needs to be supplied.")
+    if not namespace.reservationid:
+        raise CLIError("usage error: Reservation Id needs to be supplied.")
+    if not namespace.start_date:
+        raise CLIError("usage error: --start-date needs to be supplied for reservation details.")
+    if not namespace.end_date:
+        raise CLIError("usage error:--end-date needs to be supplied for reservation details.")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py]: These should not be required.  If they are correctly identified as required parameters, argparse will automatically raise the appropriate error(s)

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py [hunk 9]
```diff
@@ -7,24 +7,44 @@
...
                          exception_handler=consumption_exception_handler, validator=validate_reservations_details, client_factory=reservations_details_mgmt_client_factory)
+
+        d.custom_command('reservation id list', 'cli_consumption_list_reservations_details_by_reservation_id', transform=transform_reservation_details_list_output,
+                         exception_handler=consumption_exception_handler, validator=validate_reservations_details, client_factory=reservations_details_mgmt_client_factory)
+
+    with self.command_group('consumption pricesheet') as p:
+        p.custom_command('get', 'cli_consumption_list_pricesheet_get', transform=transform_pricesheet_get_output,
+                         exception_handler=consumption_exception_handler, validator=None, client_factory=pricesheet_mgmt_client_factory)
+
+        p.custom_command('billing period get', 'cli_consumption_list_pricesheet_by_billing_period_get', transform=transform_pricesheet_get_output,
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: `get` is not an acceptable command name in the CLI.  https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#command-naming-and-behavior-guidance

The correct name would be `show`.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py [hunk 10]
```diff
@@ -7,24 +7,44 @@
...
+from ._validators import validate_pricesheet
+from ._validators import validate_usage_bp_inputs
 
 
 def load_command_table(self, _):
     with self.command_group('consumption usage') as g:
         g.custom_command('list', 'cli_consumption_list_usage', transform=transform_usage_list_output,
                          exception_handler=consumption_exception_handler, validator=validate_both_start_end_dates, client_factory=usage_details_mgmt_client_factory)
 
+        g.custom_command('billing period list', 'cli_consumption_list_usage_by_billing_period', transform=transform_usage_list_output,
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: This looks like it creates an subgroup `billing` under `usage` that then only contains a single subgroup `period` which then contains only a single command `list`.  There should be commands under billing, and more commands under period.  If that's not the case, the command name should be changed. We should discuss.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py [hunk 11]
```diff
@@ -7,24 +7,44 @@
...
+                         exception_handler=consumption_exception_handler, validator=validate_usage_bp_inputs, client_factory=usage_details_mgmt_client_factory)
+
     with self.command_group('consumption reservations summaries') as s:
         s.custom_command('list', 'cli_consumption_list_reservations_summaries', transform=transform_reservation_summaries_list_output,
                          exception_handler=consumption_exception_handler, validator=validate_reservations_summaries, client_factory=reservations_summaries_mgmt_client_factory)
 
+        s.custom_command('reservation id list', 'cli_consumption_list_reservations_summaries_reservation_id', transform=transform_reservation_summaries_list_output,
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: Same here.

File: src/command_modules/azure-cli-consumption/setup.py [hunk 12]
```diff
@@ -33,7 +33,7 @@
 ]
 
 DEPENDENCIES = [
-    'azure-mgmt-consumption==1.1.0',
+    'azure-mgmt-consumption==2.0.0',
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/setup.py]: Is it your intention to make this module GA?  If so, you will need to schedule a dedicated command review with the CLI team.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py (hunk 11 above)
Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: 'reservation id list' is a single subgroup under consumption reservation summaries - given the best practices doc, looks like 'reservation-id-list' is a better choice?

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py (hunk 10 above)
Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: @tjprescott  intent is that - under the 'consumption usage' group there, there are 2 subgroups, 'list' and 'list by billing period', given the best practices doc, the following should work?
subgroups: 'list' and 'billing-period-list'. 
Please confirm

Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: @tjprescott  List by Billing Period expects the user to specify BillingPeriodId and the result only has items associated with that Id. Unlike 'list', list by billing period, marks billingperiodid as a required parameter. I'm not sure how we'd fold the two into a single command

Comment (williexu) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: Why could you not call one command vs the other based on whether BillingPeriodId is specified?

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py [hunk 13]
```diff
@@ -32,41 +47,68 @@ def cli_consumption_list_usage(cmd, client, billing_period_name=None, top=None,
         filter_to = "properties/usageEnd le \'{}\'".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_expression = "{} and {}".format(filter_from, filter_to)
     if top:
-        pages = client.list(scope, expand=expand, filter=filter_expression, top=top)
+        pages = client.list_by_billing_period(billing_period_name, expand=expand, filter=filter_expression, top=top)
         return list(pages.advance_page())
-    return list(client.list(scope, expand=expand, filter=filter_expression))
+    return list(client.list_by_billing_period(billing_period_name, expand=expand, filter=filter_expression))
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: `cli_consumption_list_usage()` and `cli_consumption_list_usage_by_billing_period()` essentially do the same thing. It would make much more sense to have one command that just calls the right method from the sdk depending on the presence of "billing_period_name"

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py [hunk 14]
```diff
@@ -32,41 +47,68 @@ def cli_consumption_list_usage(cmd, client, billing_period_name=None, top=None,
...
     filter_expression = None
     if start_date and end_date:
         filter_from = "properties/UsageDate ge {}".format(start_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_to = "properties/UsageDate le {}".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
         filter_expression = "{} and {}".format(filter_from, filter_to)
-        return list(client.list(scope, grain=grain, filter=filter_expression))
+        return list(client.list_by_reservation_order_and_reservation(reservation_order_id, reservation_id, grain=grain, filter=filter_expression))
 
-    return list(client.list(scope, grain=grain))
+    return list(client.list_by_reservation_order_and_reservation(reservation_order_id, reservation_id, grain=grain))
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: Same here, if "reservation_id" is None, does this essentially become the same as `client.list_by_reservation_order()`?

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py [hunk 15]
```diff
@@ -32,41 +47,68 @@ def cli_consumption_list_usage(cmd, client, billing_period_name=None, top=None,
...
+
+def cli_consumption_list_reservations_details_by_reservation_id(client, reservation_order_id, start_date, end_date, reservation_id=None):
     filter_from = None
     filter_to = None
     filter_expression = None
     filter_from = "properties/UsageDate ge {}".format(start_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
     filter_to = "properties/UsageDate le {}".format(end_date.strftime("%Y-%m-%dT%H:%M:%SZ"))
     filter_expression = "{} and {}".format(filter_from, filter_to)
-    return list(client.list(scope, filter=filter_expression))
+    return list(client.list_by_reservation_order_and_reservation(reservation_order_id, reservation_id, filter=filter_expression))
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/custom.py]: Same

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py (hunk 5 above)
Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py]: Im not sure; what I noticed from other functions here is that str() has been performed on all numeric values.

**Comment by shbha** [comment]
@williexu / @tjprescott - I've bubbled up the subgroups for each major group based on your previous comments. Thanks

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py [hunk 16]
```diff
@@ -42,36 +42,23 @@ def validate_usage_bp_inputs(namespace):
...
-        raise CLIError("usage error: --start-date needs to be supplied for reservation details.")
-    if not namespace.end_date:
-        raise CLIError("usage error:--end-date needs to be supplied for reservation details.")
-
-
-def validate_pricesheet(namespace):
-    if not namespace.billing_period_name:
-        raise CLIError("usage error: billing period name needs to be supplied.")
+    if not namespace.reservation_id:
+        raise CLIError("usage error: Reservation Id needs to be supplied.")
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py]: Then why is this not a required parameter?  This shouldn't be needed.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py (hunk 10 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]:
```Python
def list(client, billing_period=None):
  if billing_period:
    return client.list_by_billing_period(billing_period)
  return client.list()
```

**Comment by shbha** [comment]
@tjprescott - I'm assuming the last comment (2hrs ago) is related to older source code? As of last night, functionality has been folded in. Also, validators were cleaned up. If any, would you confirm which validators you think are unnecessary in the latest code?

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py [hunk 17]
```diff
@@ -27,4 +28,8 @@ def load_command_table(self, _):
 
     with self.command_group('consumption reservations details') as d:
         d.custom_command('list', 'cli_consumption_list_reservations_details', transform=transform_reservation_details_list_output,
-                         exception_handler=consumption_exception_handler, validator=validate_reservations_details, client_factory=reservations_details_mgmt_client_factory)
+                         exception_handler=consumption_exception_handler, validator=None, client_factory=reservations_details_mgmt_client_factory)
+
+    with self.command_group('consumption pricesheet') as p:
+        p.custom_command('show', 'cli_consumption_list_pricesheet_show', transform=transform_pricesheet_show_output,
+                         exception_handler=consumption_exception_handler, validator=None, client_factory=pricesheet_mgmt_client_factory)
```
Comment (williexu) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: You can remove the `validator` kwarg

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py (hunk 5 above)
Comment (williexu) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py]: What happens if the str() calls are removed?

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py (hunk 17 above)
Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: sure.. i have another PR I'll publish today, i'll make these changes then. Thanks.

Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py]: @williexu  - Regarding str conversion, if removed, json conversion fails when reading decimal or any non-string type.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py (hunk 5 above)
Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_transformers.py]: @williexu - Regarding str conversion, if removed, json conversion fails when reading decimal or any non-string type.

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_help.py [hunk 18]
```diff
@@ -34,3 +43,13 @@
     type: command
     short-summary: Show the details of Azure resource consumption, either as an invoice or within a billing period.
 """
+
+helps['consumption pricesheet'] = """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_help.py]: Is "pricesheet" one word?  If not, strongly consider hyphenating "consumption price-sheet"

File: src/command_modules/azure-cli-consumption/HISTORY.rst [hunk 19]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.2.2
++++++
+* Added new commands for budgets, marketplace and pricesheet APIs
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/HISTORY.rst]: This should be reworded because it is no longer true. I would recommend something like:
```
* Added `consumption pricesheet show` command.
* `consumption usage list`: Added support to list by billing period.
...
```

File: src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_help.py (hunk 18 above)
Comment (shbha) [reply src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_help.py]: Its not a single word, but according to the rest api published (https://docs.microsoft.com/en-us/rest/api/consumption/pricesheet/get ), its defined as a single word. So keeping with the pattern to avoid confusion.

File: src/command_modules/azure-cli-consumption/HISTORY.rst [hunk 20]
```diff
@@ -4,7 +4,7 @@ Release History
 ===============
 0.2.2
 +++++
-* Added new commands for budgets, marketplace and pricesheet APIs
+* Added new commands for pricesheet API
```
Comment (tjprescott) [inline src/command_modules/azure-cli-consumption/HISTORY.rst]: As far as I understand this still doesn't capture the scope of changes in the PR.  You added one new command and added functionality to existing commands.  That should be spelled out as specifically as possible so users don't have to hunt for the new features.

Comment (shbha) [reply src/command_modules/azure-cli-consumption/HISTORY.rst]: sure. i'll make the change.
